pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -310,6 +316,29 @@ func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, err
 	}
 
+	// Query for StorageClass for AD API if provided
```

Comment:
I'm just wondering if we want to perhaps throw an error if the user configured Ephemeral storage while filling in the storageclassname.

Reply:
was relying on the `StorageType` field to fully dictate whether to query for a StorageClass or not, which ignore other Storage related fields

Pros of **not** throwing error:
- does not block components from being deployed
Cons:
- users might  be unaware they have not set the StorageType properly (though it will be mentioned in the docs StorageType needs to be set to Persistent to query for the StorageClass) and is using Ephemeral storage


Pros of  throwing error:
- AnomalyDetection needs to be fully configured, gives users more assurance that the StorageClass is being used
Cons:
- blocks the other components from deploying

@rene-dekker let me know what you  think

Reply:
It would be a shame if the user is under the assumption that data is persistent, while it is not. That is why I thought an error would be helpful.
I am fine either way. Perhaps others have an opinion on this as well.

Reply:
> I'm just wondering if we want to perhaps throw an error if the user configured Ephemeral storage while filling in the storageclassname.

Yes, absolutely. We should validate against configuration that doesn't make it clear what the user's intent was.

Generally we should:

- Default any fields that aren't set.
- Check that the resulting object is valid and "makes sense" 

In this case, setting it to "Ephemeral" and also specifying a storage class is not self-consistent, so we should reject it and refuse to take any action until the user clarifies their intent.

Reply:
added validation https://github.com/tigera/operator/pull/2094/commits/0242a6e687adf0bac93f81213982ec3928c81dd8